scripts/lib/CIME/XML/compilerblock.py

```diff
@@ -208,8 +207,8 @@ def _add_elem_to_lists(self, name, elem, value_lists):
             value_lists[name] = PossibleValues(name, setting,
                                                self._specificity, depends)
         else:
-            value_lists[name].add_setting(setting, self._specificity,
-                                          depends)
+            specificity = 0 if len(elem.xml_element.attrib) else self._specificity
```

Comment:
Again, this was already merged, but I'm not liking this at all upon second glance. First, this is not a bug fix, but a new behavior (and one that I don't even understand the purpose of). It should probably have a test associated with it. Secondly, the order in which settings are specified in config_compilers.xml is not supposed to matter. But because this line is inside this conditional, it will never affect the first time a particular setting is specified in a block.

Reply:
I take back what I said about the testing, since I notice that the new test actually does cover this. But I will note that this XML can now fail with an ambiguity error:

```
<compiler COMPILER="intel">
  <MPIFC MPILIB="openmpi">mpifc</MPIFC>
</compiler>
<compiler MACH="mymach" COMPILER="intel">
  <MPIFC MPILIB="openmpi">mpif90</MPIFC>
</compiler>
```

Also, my comment about the order of the blocks mattering stands.